### DISCUSSION TITLE
Way to remove slide-number for n^th slide in a revealjs presentation?

### DISCUSSION THREAD

**Discussion by melissavanbussel** [quarto-dev/quarto-cli#5566] [Q&A] [answered]
I would like to remove the slide-number for the $n^{th}$ slide in a revealjs presentation, without removing the slide-number from any of the other slides. Is there a way to achieve this? 

I can get this to work for the title slide by using the solution described [here](https://stackoverflow.com/questions/75351401/remove-slide-number-and-footer-on-quarto-title-slide), but can't figure out how to apply it to a non-title slide.

Any clever workarounds / alternatives way to tackle this problem are appreciated :)

Thanks!

**Comment by mcanouil** [comment, marked as answer]
Initially from: <https://stackoverflow.com/questions/31146251/hide-slide-number-on-title-page>

````qmd
---
title: "Untitled"
format:
  revealjs:
    slide-number: true
    include-after-body: 
      - text: |
          <script type="text/javascript">
          Reveal.addEventListener('slidechanged', (event) => {
            const isSnOn = (event.currentSlide.dataset.hideSlideNumber !== 'true');
            Reveal.configure({ slideNumber: isSnOn });
          });
          </script>
---

## Slide 1

## Slide 2

## Slide 3 without slide number {data-hide-slide-number="true"}

## Slide 4 without slide number {data-hide-slide-number="true"}
````


https://github.com/quarto-dev/quarto-cli/assets/8896044/ebf44ccc-a32a-428e-9d30-19d7a6e6185b

**Reply by cscheid** [reply]
I wonder if we should make this a real quarto option..

**Reply by mcanouil** [reply]
At least for the title page, I believe this is the thing nearly all users will want (even if they don't know it).

Maybe, something like "remove all slide number in slides after" an `.appendix` class.

**Reply by melissavanbussel** [reply]
This is great, thank you so much @mcanouil! 

Is there a way to apply the `{data-hide-slide-number='true'}` to the title slide as well?

**Reply by mcanouil** [reply]
There are various ways, but here I simply add a new JavaScript event listener to hide the slide number on title page when all is loaded.

````qmd
---
title: "Untitled"
format:
  revealjs:
    slide-number: true
    theme: no-slide-number.scss
    include-after-body: 
      - text: |
          <script type="text/javascript">
          Reveal.addEventListener('slidechanged', (event) => {
            const isSnOn = (event.currentSlide.dataset.hideSlideNumber !== 'true');
            Reveal.configure({ slideNumber: isSnOn });
          });
          Reveal.addEventListener('ready', (event) => {
            const isFirst = event.indexh === 0;
            Reveal.configure({ slideNumber: !isFirst });
          });
          </script>
---

## Slide 1

## Slide 2

## Slide 3 without slide number {data-hide-slide-number="true"}

## Slide 4 without slide number {data-hide-slide-number="true"}
````

**Reply by cderv** [reply]
I think this is a feature that should be possible upstream, but still a feature request for now: https://github.com/hakimel/reveal.js/issues/3151

I think it would be worth adding a feature for this in Quarto though.

**Reply by mcanouil** [reply]
Additionally, there is a CSS trick to hide slide number on the title slide:

```css
.reveal .slide-number > a[href*='title-slide'] {
  display: none;
}
```

There is also another JavaScript to do that as highlighted by @cderv in #557

**Comment by mcanouil** [comment]
FYI:
- Opened issue:
  - #1445
- Related discussions:
  - #557
  - #825
  - #4181
  - #5566

**Comment by grantmcdermott** [comment]
It needs to be added to every slide that you'd like to avoid counting for, but I suppose `visibility="uncounted"` is a canonical solution now?

https://quarto.org/docs/presentations/revealjs/advanced.html#uncounted-slides
